File: frontend/src/types/Release.ts
import type { TaskBlocker, GroundTask } from "./GroundTask";
import type { DelayBlocker } from "./DelayEvent";
import type { BookingBlocker, ResourceBooking } from "./ResourceBooking";
import type { FlightTurnaroundDetail } from "./FlightTurnaround";

export interface ReleaseBlockers {
  tasks: TaskBlocker[];
  delays: DelayBlocker[];
  bookings: BookingBlocker[];
}

export interface ReleaseResponse {
  message: string;
  turnaround: FlightTurnaroundDetail;
  tasks: GroundTask[];
  bookings: ResourceBooking[];
  released_count: number;
}

export interface ReleaseErrorPayload {
  code: string;
  message: string;
  blockers?: ReleaseBlockers;
}

File: backend/src/types/Release.go
package types

import (
	"net/http"

	"groundTurn/src/models"
)

type ReleaseBlockers struct {
	Tasks    []TaskBlocker    `json:"tasks"`
	Delays   []DelayBlocker   `json:"delays"`
	Bookings []BookingBlocker `json:"bookings"`
}

type ReleaseResponse struct {
	Message       string                   `json:"message"`
	Turnaround    FlightTurnaroundDetail   `json:"turnaround"`
	Tasks         []models.GroundTask      `json:"tasks"`
	Bookings      []models.ResourceBooking `json:"bookings"`
	ReleasedCount int                      `json:"released_count"`
}

type ReleaseRejectedResponse struct {
	Code     string          `json:"code"`
	Message  string          `json:"message"`
	Blockers ReleaseBlockers `json:"blockers"`
	HTTPCode int             `json:"-"`
}

func NewReleaseRejected(message string, blockers ReleaseBlockers) *ReleaseRejectedResponse {
	return &ReleaseRejectedResponse{Code: "RELEASE_BLOCKED", Message: message, Blockers: blockers, HTTPCode: http.StatusConflict}
}
